src/palace/manager/celery/tasks/axis.py

```diff
@@ -0,0 +1,404 @@
+    service_name: str,
+    collection: Collection,
+    default_counter: int = 0,
+) -> Timestamp:
+    """Find or create a Timestamp.
+    A new timestamp will have .finish set to None, since the first
+    run is presumably in progress.
+    """
+    timestamp, new = get_one_or_create(
+        _db,
+        Timestamp,
+        service=service_name,
+        service_type=Timestamp.TASK_TYPE,
+        collection=collection,
+        create_method_kwargs=dict(
+            start=default_start_time,
+            finish=None,
+            counter=default_counter,
+        ),
+    )
+    return timestamp
+
+
+@shared_task(queue=QueueNames.default, bind=True, max_retries=4)
+def import_identifiers(
+    task: Task,
+    collection_id: int,
+    identifiers: list[str] | None,
+    processed_count: int = 0,
+    batch_size: int = DEFAULT_BATCH_SIZE,
+    target_max_execution_time_in_seconds: float = TARGET_MAX_EXECUTION_SECONDS,
+) -> None:
+    """
+    This method creates new or updates new editions and license pools for each identifier in the list of identifiers.
+    It will query the axis availability api in batches of {batch_size} IDs and process each result in a single database
+    transaction.  It will continue in this way until it has finished the list or exceeded the max execution time
+    which defaults to 2 minutes.  If it has not finished in the target time, it will requeue the task with the
+    remaining unprocessed identifiers.
+    """
+    with task.transaction() as session:
+        collection = Collection.by_id(session, id=collection_id)
+        if not collection:
+            task.log.error(f"Collection not found:  {collection_id} : ignoring...")
+            return None
+
+        def log_run_end_message() -> None:
+            task.log.info(
+                f"Finished importing identifiers for collection ({collection.name}, id={collection_id}), "
+                f"task(id={task.request.id})"
+            )
+
+        if identifiers is None:
+
+            task.log.info(
+                f"Identifiers list is None: the list_identifiers_for_import "
+                f"must have been locked. Ignoring import run for collection_id={collection_id}"
+            )
+            return
+
+        if not identifiers:
+            task.log.info(
+                f"Identifiers list is empty: Nothing remains for processing in task(id={task.request.id})"
+            )
+            log_run_end_message()
+            return
+
+        api = create_api(session=session, collection=collection)
+        start_seconds = time.perf_counter()
+        total_imported_in_current_task = 0
+        while len(identifiers) > 0:
+            batch = identifiers[:batch_size]
+
+            try:
+                for metadata, circulation in api.availability_by_title_ids(
+                    title_ids=batch
+                ):
+                    process_book(task, session, api, metadata, circulation)
+            except Exception as e:
```

Comment:
Minor: It would be nice to narrow this a bit instead of catching every exception. We were handling `ObjectDeletedError`, `StaleDataError` before, can we just catch those here?